File: Objective-C/CustomActivityIndicator/ViewController.m
//
//  ViewController.m
//  CustomActivityIndicator
//
//  Created by Dave on 5/25/18.
//  Copyright © 2018 High Tree Development. All rights reserved.
//

#import "ViewController.h"
#import "CustomActivityIndicator.h"


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.defaultButton.layer.borderWidth = 1.0;
    self.defaultButton.layer.borderColor = UIColor.whiteColor.CGColor;
    self.defaultButton.layer.cornerRadius = 4.0;
    
    self.animateButton.layer.borderWidth = 1.0;
    self.animateButton.layer.borderColor = UIColor.whiteColor.CGColor;
    self.animateButton.layer.cornerRadius = 4.0;
    
    self.labelButton.layer.borderWidth = 1.0;
    self.labelButton.layer.borderColor = UIColor.whiteColor.CGColor;
    self.labelButton.layer.cornerRadius = 4.0;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)showPressed:(id)sender {
    [CustomActivityIndicator.shared show:self.view];
    [CustomActivityIndicator.shared show:self.view backgroundColor:UIColor.darkGrayColor size:80.0 duration:1.0];
    
    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2.0 * NSEC_PER_SEC));
    dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
        [CustomActivityIndicator.shared hide:self.view];
    });
}

- (IBAction)showLabelPressed:(id)sender {
    [CustomActivityIndicator.shared show:self.view backgroundColor:UIColor.darkGrayColor textColor:UIColor.whiteColor labelText:@"Loading user data" duration:1.0];
    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2.0 * NSEC_PER_SEC));
    dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
        [CustomActivityIndicator.shared hide:self.view duration:1.0];
    });
}

- (IBAction)animatePressed:(id)sender {
    [CustomActivityIndicator.shared show:self.view backgroundColor:UIColor.darkGrayColor size:80.0 duration:1.0];
    
    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2.0 * NSEC_PER_SEC));
    dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
        [CustomActivityIndicator.shared hide:self.view duration:1.0];
    });
}

@end

File: CustomActivityIndicator/ViewController.swift
//
//  ViewController.swift
//  CustomActivityIndicator
//
//  Created by Dave on 1/24/18.
//  Copyright © 2018 High Tree Development. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var button: UIButton!
    @IBOutlet weak var colorButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        button.layer.borderWidth = 1
        button.layer.borderColor = UIColor.white.cgColor
        button.layer.cornerRadius = 4
        
        colorButton.layer.borderWidth = 1
        colorButton.layer.borderColor = UIColor.white.cgColor
        colorButton.layer.cornerRadius = 4
    }
    
    @IBAction func buttonPressed(_ sender: Any) {
        CustomActivityIndicator.shared.showActivityIndicator(uiView: self.view)
        CustomActivityIndicator.shared.hideActivityIndicator(uiView: self.view, delay: 1.5)
    }
    
    @IBAction func colorPressed(_ sender: Any) {
        CustomActivityIndicator.shared.showActivityIndicator(uiView: self.view, labelText: "Creating user account")
        CustomActivityIndicator.shared.hideActivityIndicator(uiView: self.view, delay: 1.5)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

}

